### ISSUE TITLE
[Hermes] Can't build a new project for dev

### ISSUE THREAD

**Issue by bartlomein** [callstack/react-native-visionos#128]
### Description

Was having issues with my build so I created a brand new one and tried to run it and getting the same error:
```
ld: building for 'visionOS-simulator', but linking in dylib (/Users/bart/dev/exampleapp/visionos/Pods/hermes-engine/destroot/Library/Frameworks/xros/hermes.framework/hermes) built for 'macOS'
clang: error: linker command failed with exit code 1 (use -v to see invocation)


** BUILD FAILED **```

### Steps to reproduce

Follow instructions to create a new app, and run it locally without any changes.

### React Native Version

0.73.3

### Affected Platforms

Other (please specify)

### Areas

Other (please specify)

### Output of `npx react-native info`

```text
none
```

### Stacktrace or Logs

```text
none
```

### Reproducer

none

**Comment by okwasniewski** [comment]
Hey, are you running latest version of cmake? Can you also try the latest version of react native visionOS (0.73.7)?

**Comment by bartlomein** [comment]
Hmm I updated both of those and ran into another issue 
```** BUILD FAILED **


The following build commands failed:
	SwiftEmitModule normal x86_64 Emitting\ module\ for\ mmvisionpro (in target 'mmvisionpro' from project 'mmvisionpro')
(1 failure)```

At one point it did build with no errors, just when a simulator was not selected, just a Generic Vision Pro device. I have a feeling I might have messed up some XCode setting.

**Comment by bartlomein** [comment]
Actually I think I got it!
Had to add hermes to Frameworks, libraries and Embedded content here:

<img width="709" alt="Screenshot 2024-03-10 at 3 42 46 PM" src="https://github.com/callstack/react-native-visionos/assets/3741391/1a186073-8cf5-4bf0-9d70-cc01fa663fae">

**Comment by josh-deprogram** [comment]
I'm hitting the same problem...

CMake - 3.28.3
RN-VisionOS - 0.73.7
XCode - 15.3
VisionOS - 1.1

Adding `libReact-hermes.a` to Frameworks doesn't seem to resolve either

**Comment by josh-deprogram** [comment]
Aha! I have it compiling, disabled hermes from pods and removed the framework in Xcode..
For anyone else kicking the compiler try
`:hermes_enabled => false,`

**Comment by robertherber** [comment]
@josh-deprogram Thanks, also got it running by disabling Hermes for now 👍

**Comment by jariwalabhavesh** [comment]
@josh-deprogram Thanks, I had disable hermes in root ios and visionos podfile and it start working.
Thanks

**Comment by okwasniewski** [comment]
Can you guys try out RC release? https://github.com/callstack/react-native-visionos/releases/tag/v0.74.0-rc.0-visionos

**Comment by 15bedirhan** [comment]
I tried out RC release version but still have same error. Any updates on this?

<img width="843" alt="Screenshot 2024-03-27 at 15 05 50" src="https://github.com/callstack/react-native-visionos/assets/73651069/ace96287-9266-4f73-9064-70d466a4a3d0">

**Comment by okwasniewski** [comment]
@15bedirhan Do you have the latest version of cmake? (3.29.0 or 3.28.4)?

**Comment by 15bedirhan** [comment]
> @15bedirhan Do you have the latest version of cmake? (3.29.0 or 3.28.4)?

Yes my cmake version is 3.29.0. 

For now I continue disabled hermes in visionos/podfile. It works after that
<img width="490" alt="Screenshot 2024-03-27 at 15 30 09" src="https://github.com/callstack/react-native-visionos/assets/73651069/e9e43259-8504-4d74-b232-520b3fdf7a2c">

**Comment by okwasniewski** [comment]
Okay I did some investigation together with @thiagobrez, looks like it's [`Logging.cpp`](https://github.com/facebook/hermes/blob/1d79636164f6a0dbdcffa4e72e14f1499955b73f/lib/Platform/Logging.cpp#L33) file from Hermes which causes those issues for some people. It must be some missing tool/library as its not happening on my Mac.

**Comment by thiagobrez** [comment]
Workaround for now:
- Double click the error line `use of undeclared identifier 'vsnprintf'`
- Logging.cpp from Hermes will open
- Comment everything inside the function

**Comment by okwasniewski** [comment]
The fix is coming: https://github.com/facebook/hermes/pull/1358

**Comment by okwasniewski** [comment]
I'm disabling Hermes by default until the above PR gets merged. If you still want to use Hermes remove `:hermes_enabled => false` from `visionos/Podfile`.  And do what @thiagobrez explained above. The fix should land shortly.

**Comment by matthargett** [comment]
upstream PR merged, so this can be revisited for the next release of react-native-visionos

**Comment by okwasniewski** [comment]
Hey, 

This issue should now be resolved in React Native 0.74.3. Backport to 0.73.x will be available once this PR get's merged: https://github.com/facebook/hermes/pull/1432

**Comment by bgazzera** [comment]
For others who may also experience this error, we also came across this issue while using  CMake version 3.28.0, recommended in the README.md. 

Commenting out Hermes / Adding libReact-hermes.a didn't work for us.
Fixed by updating CMake to 3.29.9 or 3.30.0

@okwasniewski if it helps, I can submit a PR to update the README and also the docs repo. And thanks for all your work, great stuff.

**Comment by okwasniewski** [comment]
> For others who may also experience this error, we also came across this issue while using  CMake version 3.28.0, recommended in the README.md. 
> 
> 
> 
> Commenting out Hermes / Adding libReact-hermes.a didn't work for us.
> 
> Fixed by updating CMake to 3.29.9 or 3.30.0
> 
> 
> 
> @okwasniewski if it helps, I can submit a PR to update the README and also the docs repo. And thanks for all your work, great stuff.

It would be great if you could create a PR, I forgot to update it.
